plugins/lua/sort/places.lua

```diff
@@ -0,0 +1,121 @@
+    if zone.name == nil or zone.name == '' then
+        table.insert(result, get_default_zone_name(zone.type))
+    else
+        table.insert(result, zone.name)
+    end
+
+    -- allow zones w/ assignments to be searchable by their assigned unit
+    if zone.assigned_unit ~= nil then
+        table.insert(result, dfhack.TranslateName(zone.assigned_unit.name))
+        table.insert(result, dfhack.units.getReadableName(zone.assigned_unit))
+    end
+
+    -- allow zones to be searchable by type
+    if zone.location_id == -1 then -- zone is NOT a special location and we don't need to do anything special for type searching
+        table.insert(result, df.civzone_type[zone.type]);
+    else -- zone is a special location and we need to get its type from world data
+        table.insert(result, dfhack.TranslateName(site.buildings[zone.location_id].name, true))
+        table.insert(result, df.civzone_type[site.buildings[zone.location_id].name.type])
+    end
+
+    -- allow barracks to be searchable by assigned squad
+    for _, squad in ipairs(zone.squad_room_info) do
+        table.insert(result, dfhack.military.getSquadName(squad.squad_id))
+    end
+
+    return table.concat(result, ' ')
+end
+
+-- ----------------------
+-- PlacesOverlay
+--
+
+PlacesOverlay = defclass(PlacesOverlay, sortoverlay.SortOverlay)
+PlacesOverlay.ATTRS{
+    default_pos={x=71, y=8},
+    viewscreens='dwarfmode/Info',
+    frame={w=40, h=6}
+}
+
+function PlacesOverlay:init()
+    self:addviews{
+        widgets.BannerPanel{
+            frame={l=0, t=0, r=0, h=1},
+            visible=self:callback('get_key'),
+            subviews={
+                widgets.EditField{
+                    view_id='search',
+                    frame={l=1, t=0, r=1},
+                    label_text="Search: ",
+                    key='CUSTOM_ALT_S',
+                    on_change=function(text) self:do_search(text) end,
+                },
+            },
+        },
+    }
+
+    self:register_handler('ZONES', buildings.list[df.buildings_mode_type.ZONES], curry(sortoverlay.single_vector_search, {get_search_key_fn=get_zone_search_key}))
+end
+
+function PlacesOverlay:get_key()
+    if info.current_mode == df.info_interface_mode_type.BUILDINGS then
+        -- TODO: Replace nested if with 'return df.buildings_mode_type[buildings.mode]' once other handlers are written
+        -- Not there right now so it doesn't render a search bar on unsupported Places subpages
+        if buildings.mode == df.buildings_mode_type.ZONES then
+            return 'ZONES'
+        end
+    end
+end
+
+function PlacesOverlay:onRenderBody(dc)
+    PlacesOverlay.super.onRenderBody(self, dc)
+    if self.refresh_search then
+        self.refresh_search = nil
+        self:do_search(self.subviews.search.text)
+    end
+end
+
+function PlacesOverlay:onInput(keys)
+    if keys._MOUSE_L then
+        self.refresh_search = true
+    end
+    return PlacesOverlay.super.onInput(self, keys)
+end
```

Comment:
is this necessary on these tabs? The WORK_DETAILS screen needs this because clicking on any of the buttons refreshes the underlying vectors, which then need to be re-filtered. I don't think that situation applies to the Places subtabs (does it?)

Reply:
I copied this from UnitSelectorOverlay -- but no, the vector's don't need to be reloaded when any buttons are clicked. Will remove and make sure everything still works.